packages/react/src/primitives/PhoneNumberField/utils.ts

```diff
@@ -0,0 +1,46 @@
+export const formatPhoneNumber = (
+  dialCode: string,
+  phoneNumberString: string
+): string => {
+  const cleaned = ('' + phoneNumberString).replace(/\D/g, '');
+  let formatted: string;
+  switch (dialCode) {
+    case '+1':
+      formatted = cleaned.replace(/(\d{3})(\d{3})(\d{4})/, '($1) $2-$3');
+      break;
+    case '+7':
+      formatted = cleaned.replace(/(\d{3})(\d{3})(\d{4})/, '($1) $2-$3');
+      break;
+    case '+33':
+      formatted = cleaned.replace(
+        /(\d{2})(\d{2})(\d{2})(\d{2})(\d{2})/,
+        '$1 $2 $3 $4 $5'
+      );
+      break;
+    case '+39':
+      formatted = cleaned.replace(/(\d{4})(\d{4})(\d{3})/, '$1 $2 $3');
+      break;
+    case '+44':
+      formatted = cleaned.replace(/(\d{4})(\d{3})(\d{4})/, '$1 $2-$3');
+      break;
+    case '+49':
+      formatted = cleaned.replace(/(\d{4})(\d{3})(\d{4})/, '$1 $2 $3');
+      break;
+    case '+52':
+      formatted = cleaned.replace(/(\d{3})(\d{3})(\d{4})/, '$1 $2 $3');
+      break;
+    case '+81':
+      formatted = cleaned.replace(/(\d{3})(\d{4})(\d{4})/, '$1 $2 $3');
+      break;
+    case '+91':
+      formatted = cleaned.replace(/(\d{5})(\d{5})/, '$1 $2');
+      break;
+    case '+234':
+      formatted = cleaned.replace(/(\d{3})(\d{4})(\d{4})/, '$1 $2 $3');
+      break;
+    // Add more cases for other country codes as needed
```

Comment:
How many country codes should we support?

Reply:
The related GItHub issue did not specify the amount of dial codes. Currently I've added some regex logic to support 10 dial codes, we can add more if needed OR we can use the libphonenumber-js library, so we don't need to add manually every single dial code, however I'm not entirely sure if that library won't cause any runtime failures or/and require additional testing.

Reply:
Unit tests have been added